plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.data.ViewportDataSource
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportStateChangedObserver
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+
+/**
+ * Mapbox default implementation for [ViewportPlugin].
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private var currentState: ViewportState = ViewportState.IDLE
+  override val state: ViewportState
+    get() = currentState
+
+  override fun getDataSource(state: ViewportState): ViewportDataSource {
+    TODO("Not yet implemented")
+  }
+
+  override fun requestCameraToState(
+    state: ViewportState,
+    stateTransitionOptionsBlock: ViewportTransitionOptions.Builder.() -> Unit,
```

Comment:
`stateTransitionOptionsBlock` is not used.

Reply:
This PR just prototypes public interface and left the implementation details 😄 We will continue implementation when we think the public interfaces are good enough.